resources/assets/coffee/react/beatmapset-page/success-rate.coffee

```diff
@@ -0,0 +1,86 @@
+###
+# Copyright 2016 ppy Pty. Ltd.
+#
+# This file is part of osu!web. osu!web is distributed with the hope of
+# attracting more community contributions to the core ecosystem of osu!.
+#
+# osu!web is free software: you can redistribute it and/or modify
+# it under the terms of the Affero GNU General Public License version 3
+# as published by the Free Software Foundation.
+#
+# osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+{div, p, span} = React.DOM
+el = React.createElement
+
+legend = ['retry', 'fail']
+
+class BeatmapsetPage.SuccessRate extends React.Component
+  componentDidMount: ->
+    @_renderChart()
+
+  componentDidUpdate: ->
+    @_renderChart()
+
+  componentWillUnmount: ->
+    $(window).off '.beatmapSetPageSuccessRate'
+
+  _renderChart: ->
+    return unless @props.failtimes.length > 0
+
+    failtimes = _.map @props.failtimes, (m) -> m.data
+    data = _.zip failtimes[0], failtimes[1]
```

Comment:
with just two rows, I think it's better to sort them here. Even better if you can somehow infer the class names using `fail` and `retry` instead of `0` and `1`.